source/GameAssets.h

```diff
@@ -0,0 +1,76 @@
+/* GameAssets.h
+Copyright (c) 2022 by quyykk
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef GAME_ASSETS_H_
+#define GAME_ASSETS_H_
+
+#include "SoundSet.h"
+#include "SoundQueue.h"
+#include "SpriteSet.h"
+#include "SpriteQueue.h"
+#include "UniverseObjects.h"
+
+#include <future>
+#include <map>
+#include <string>
+#include <vector>
+
+
+
+// This class contains the game assets needed to play the game: The game data, the images and the sounds.
+class GameAssets {
+	// GameData currently is the orchestrating controller for all game definitions.
+	friend class GameData;
+public:
+	static constexpr int None = 0x0;
+	static constexpr int Debug = 0x1;
+	static constexpr int OnlyData = 0x2;
+
+
+public:
+	GameAssets() noexcept;
+
+	// Asset objects cannot be copied.
+	GameAssets(const GameAssets &) = delete;
+	GameAssets(GameAssets &&) = delete;
+	GameAssets &operator=(const GameAssets &) = delete;
+	GameAssets &operator=(GameAssets &&) = delete;
+
```

Comment:
To clarify, when one sees a class that needs to restrict copies and moves, and also has a custom default constructor, one wonders if there is supposed to be a user-defined destructor as well.

We know the compiler generated destructor is doing the right thing in this case, but only because we've gone and inspected this class's members.

What's nice is that this code smell will vanish once the asset loading queues are moved to a more appropriate owning scope.

Reply:
Fair enough. TBH I'm not quite happy with how the queues work. It makes ownership and related stuff complicated.

So, I was planning on doing (yet) another PR that would generalize them as a TaskQueue class that can execute arbitrary tasks. Such a class would be "static" so no worries about who owns what. Thoughts? (It's fine if you don't have any, you'll see the changes in that PR).